win32ss/gdi/ntgdi/font.c

```diff
@@ -464,19 +463,19 @@ NtGdiAddFontResourceW(
     if ((cwc <= 1) || (cwc > UNICODE_STRING_MAX_CHARS))
         return 0;
 
-    SafeFileName.MaximumLength = (USHORT)(cwc * sizeof(WCHAR));
-    SafeFileName.Length = SafeFileName.MaximumLength - sizeof(UNICODE_NULL);
+    SafeFileName.Length = (USHORT)((cwc - 1) * sizeof(WCHAR));
+    SafeFileName.MaximumLength = SafeFileName.Length + 2 * sizeof(UNICODE_NULL); // Security issue
     SafeFileName.Buffer = ExAllocatePoolWithTag(PagedPool,
                                                 SafeFileName.MaximumLength,
                                                 TAG_STRING);
     if (!SafeFileName.Buffer)
-    {
         return 0;
-    }
 
     _SEH2_TRY
     {
         ProbeForRead(pwcFiles, cwc * sizeof(WCHAR), sizeof(WCHAR));
+        if (pwcFiles[cwc - 1] != UNICODE_NULL)
```

Comment:
The name "pwcFiles" plural seems to implicitly indicate this is indeed a multi-string passed there, so the double-NUL termination you claim it's for "security issue" is not for pseudosecurity purposes, but is there for a good reason.